### PR TITLE
swarm/storage/feeds: Renamed MRU to Swarm Feeds

### PR DESCRIPTION
**Pull request by jpeletier**
Renaming MRUs to Swarm Feeds
===

## Abstract
As the Mutable Resource Updates code evolved in the last couple of months, we realized that the end result looked more like a user posting to an update feed than a user "updating a previously created resource".

This PR looks at all Swarm code in excruciating detail, renaming all identifiers and comments so that it reflects this paradigm.

There are no functionality changes in this PR.

## Terms that have been renamed:

* `mru` package is now `feed` and it is located in `/swarm/storage/feed`
* `mru.View` is renamed to `feed.Feed`
* `/bzz-resource:` endpoint is renamed to `/bzz-feed:`
* Swarm command-line `swarm resource` has been renamed to `swarm feed`.

## What objects can I work with from the `feed` package?
You will only need to use three objects:
* `Feed`
* `Query`
* `Request`

### What is a `Feed`?
You can think of a `Feed` as a user's Twitter account, where he/she posts updates about a particular `Topic` (hashtag). In fact, the `Feed` object is simply defined as:
```go
type Feed struct {
	Topic Topic
	User  common.Address
}
```
Users can post about any topic or have topics in common. If you know the user's address and agree on a particular topic, you can then effectively "follow" that user's feed.

### What is a `Query`?
The `Query` object allows you to build a query to browse a particular `Feed`. The default `Query`, obtained with `feed.NewQueryLatest()` will build a `Query` that retrieves the latest update of the given `Feed`.
You can also use `feed.NewQuery()` instead, if you want to build a `Query` to look up an update before a certain date.
Advanced usage of `Query` includes hinting the lookup algorithm for faster lookups. The default hint (`lookup.NoClue`) will have your node track feeds you query frequently and handle hints automatically.

### What is a `Request`?
The `Request` object makes it easy to construct and sign a request to Swarm to update a particular `Feed`. It contains methods to sign and add data.
A `Request` can also be serialized to JSON in case you need your application to delegate signatures, such as having a browser sign a feed update request.

### Example client code:
See for yourself how the new naming improves readability of your code. The following example uses Swarm's client package to retrieve a ready-to-sign `Request`, add data, sign it and post it to Swarm:

```go
// Build a `Feed` object to track a particular user's updates
f := new(feed.Feed)
f.User = signer.Address()
f.Topic, _ = feed.NewTopic("weather",nil)

// Build a `Query` to retrieve a current Request for this feed
query := feeds.NewQueryLatest(&f, lookup.NoClue)

// Retrieve a ready-to-sign request using our query
// (queries can be reused)
request, err := client.GetFeedRequest(query, "")
if err != nil {
    utils.Fatalf("Error retrieving feed status: %s", err.Error())
}

// set the new data
request.SetData([]byte("Weather looks bright and sunny today, we should merge this PR and go out enjoy"))

// sign update
if err = request.Sign(signer); err != nil {
    utils.Fatalf("Error signing feed update: %s", err.Error())
}

// post update
err = client.UpdateFeed(request)
if err != nil {
    utils.Fatalf("Error updating feed: %s", err.Error())
    return
}
```

### CODE REVIEW

File: swarm/OWNERS [hunk 1]
```diff
@@ -22,5 +22,5 @@ swarm
 ├── storage ─────────────── ethersphere
 │   ├── encryption ──────── @gbalint, @zelig, @nagydani
 │   ├── mock ────────────── @janos
-│   └── mru ─────────────── @nolash
+│   └── feeds ───────────── @nolash
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply swarm/OWNERS]: :) what an honor.
Added. Thanks.

File: swarm/storage/feeds/handler_test.go [hunk 2]
```diff
@@ -266,11 +266,11 @@ func TestSparseUpdates(t *testing.T) {
 	defer teardownTest()
 	defer os.RemoveAll(datadir)
 
-	// create a new resource
+	// create a new Feed
```
Comment (zelig) [inline swarm/storage/feeds/handler_test.go]: feed

File: swarm/storage/feeds/handler_test.go [hunk 3]
```diff
@@ -382,7 +382,7 @@ func TestValidator(t *testing.T) {
 }
 
 // tests that the content address validator correctly checks the data
-// tests that resource update chunks are passed through content address validator
+// tests that Feed update chunks are passed through content address validator
```
Comment (zelig) [inline swarm/storage/feeds/handler_test.go]: feed

File: swarm/storage/feeds/handler_test.go [hunk 4]
```diff
@@ -451,20 +451,20 @@ func TestValidatorInStore(t *testing.T) {
...
 	err = store.Put(context.Background(), uglyChunk)
 	if err != nil {
-		t.Fatalf("expected no error on resource update chunk with resource validator only, but got: %s", err)
+		t.Fatalf("expected no error on feed update chunk with feed update validator only, but got: %s", err)
 	}
 }
 
-// create rpc and resourcehandler
-func setupTest(timeProvider timestampProvider, signer Signer) (rh *TestHandler, datadir string, teardown func(), err error) {
+// create rpc and Feeds Handler
```
Comment (zelig) [inline swarm/storage/feeds/handler_test.go]: feeds handler

File: swarm/storage/feeds/lookup/lookup.go [hunk 5]
```diff
@@ -15,7 +15,7 @@
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
 /*
-Package lookup defines resource lookup algorithms and provides tools to place updates
+Package lookup defines Feed lookup algorithms and provides tools to place updates
```
Comment (zelig) [inline swarm/storage/feeds/lookup/lookup.go]: feed

File: swarm/storage/feeds/request.go [hunk 6]
```diff
@@ -24,15 +24,15 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/swarm/storage"
-	"github.com/ethereum/go-ethereum/swarm/storage/mru/lookup"
+	"github.com/ethereum/go-ethereum/swarm/storage/feeds/lookup"
 )
 
-// Request represents an update and/or resource create message
+// Request represents a request to sign or signed Feed Update message
```
Comment (zelig) [inline swarm/storage/feeds/request.go]: feed update

File: swarm/storage/feeds/request_test.go [hunk 7]
```diff
@@ -47,51 +47,49 @@ func areEqualJSON(s1, s2 string) (bool, error) {
 }
 
 // TestEncodingDecodingUpdateRequests ensures that requests are serialized properly
-// while also checking cryptographically that only the owner of a resource can update it.
+// while also checking cryptographically that only the owner of a Feed can update it.
```
Comment (zelig) [inline swarm/storage/feeds/request_test.go]: feed

File: swarm/storage/feeds/request_test.go [hunk 8]
```diff
@@ -138,7 +136,7 @@ func TestEncodingDecodingUpdateRequests(t *testing.T) {
 		t.Fatal("Expected DecodeUpdateRequest to fail when trying to interpret a corrupt message with an invalid signature")
 	}
 
-	// Now imagine Bob wants to create an update of his own about the same resource,
+	// Now imagine Bob wants to create an update of his own about the same Feed,
```
Comment (zelig) [inline swarm/storage/feeds/request_test.go]: feed

File: swarm/storage/feeds/request_test.go [hunk 9]
```diff
@@ -230,7 +228,7 @@ func TestUpdateChunkSerializationErrorChecking(t *testing.T) {
 	var recovered Request
 	recovered.fromChunk(chunk.Address(), chunk.Data())
 	if !reflect.DeepEqual(recovered, r) {
-		t.Fatal("Expected recovered SignedResource update to equal the original one")
+		t.Fatal("Expected recovered Request update to equal the original one")
```
Comment (zelig) [inline swarm/storage/feeds/request_test.go]: feed update request

File: swarm/storage/feeds/request_test.go [hunk 10]
```diff
@@ -250,23 +248,23 @@ func TestReverse(t *testing.T) {
 	// signer containing private key
 	signer := newAliceSigner()
 
-	// set up rpc and create resourcehandler
+	// set up rpc and create Feeds handler
```
Comment (zelig) [inline swarm/storage/feeds/request_test.go]: feeds

File: swarm/storage/feeds/sign.go [hunk 11]
```diff
@@ -28,7 +28,7 @@ const signatureLength = 65
 // Signature is an alias for a static byte array with the size of a signature
 type Signature [signatureLength]byte
 
-// Signer signs Mutable Resource update payloads
+// Signer signs Feed update payloads
```
Comment (zelig) [inline swarm/storage/feeds/sign.go]: feed

File: swarm/storage/feeds/sign.go [hunk 12]
```diff
@@ -65,7 +65,7 @@ func (s *GenericSigner) Address() common.Address {
 	return s.address
 }
 
-// getUserAddr extracts the address of the resource update signer
+// getUserAddr extracts the address of the Feed update signer
```
Comment (zelig) [inline swarm/storage/feeds/sign.go]: feed

File: cmd/swarm/main.go [hunk 13]
```diff
@@ -346,62 +346,62 @@ func init() {
...
 			Subcommands: []cli.Command{
 				{
-					Action:             resourceCreate,
+					Action:             feedCreateManifest,
 					CustomHelpTemplate: helpTemplate,
 					Name:               "create",
-					Usage:              "creates and publishes a new Mutable Resource manifest",
-					Description: `creates and publishes a new Mutable Resource manifest pointing to a specified user's updates about a particular topic.
-					The resource topic can be built in the following ways:
+					Usage:              "creates and publishes a new Feed manifest",
```
Comment (zelig) [inline cmd/swarm/main.go]: new feed

File: cmd/swarm/main.go [hunk 14]
```diff
@@ -346,62 +346,62 @@ func init() {
...
+					Description: `creates and publishes a new Feed manifest pointing to a specified user's updates about a particular topic.
```
Comment (zelig) [inline cmd/swarm/main.go]: new feed

File: cmd/swarm/main.go [hunk 15]
```diff
@@ -346,62 +346,62 @@ func init() {
...
 				},
 				{
-					Action:             resourceInfo,
+					Action:             feedInfo,
 					CustomHelpTemplate: helpTemplate,
 					Name:               "info",
-					Usage:              "obtains information about an existing Mutable Resource",
-					Description: `obtains information about an existing Mutable Resource
+					Usage:              "obtains information about an existing Swarm Feed",
+					Description: `obtains information about an existing Swarm Feed
```
Comment (zelig) [inline cmd/swarm/main.go]: Swarm feed

File: swarm/api/api.go [hunk 16]
```diff
@@ -956,94 +957,93 @@ func (a *API) BuildDirectoryTree(ctx context.Context, mhash string, nameresolver
 	return addr, manifestEntryMap, nil
 }
 
-// ResourceLookup finds mutable resource updates at specific periods and versions
-func (a *API) ResourceLookup(ctx context.Context, query *mru.Query) ([]byte, error) {
-	_, err := a.resource.Lookup(ctx, query)
+// FeedsLookup finds Swarm Feeds Updates at specific points in time, or the latest update
```
Comment (zelig) [inline swarm/api/api.go]: Swarm feed updates

File: swarm/api/api.go [hunk 17]
```diff
@@ -956,94 +957,93 @@ func (a *API) BuildDirectoryTree(ctx context.Context, mhash string, nameresolver
...
 	if err != nil {
 		return nil, err
 	}
 	return data, nil
 }
 
-// ResourceNewRequest creates a Request object to update a specific mutable resource
-func (a *API) ResourceNewRequest(ctx context.Context, view *mru.View) (*mru.Request, error) {
-	return a.resource.NewRequest(ctx, view)
+// FeedsNewRequest creates a Request object to update a specific Feed
```
Comment (zelig) [inline swarm/api/api.go]: specific feed

File: swarm/api/api.go [hunk 18]
```diff
@@ -956,94 +957,93 @@ func (a *API) BuildDirectoryTree(ctx context.Context, mhash string, nameresolver
...
-	return a.resource.Update(ctx, request)
+// FeedsUpdate publishes a new update on the given Feed
+func (a *API) FeedsUpdate(ctx context.Context, request *feeds.Request) (storage.Address, error) {
+	return a.feeds.Update(ctx, request)
 }
 
-// ResourceHashSize returned the size of the digest produced by the Mutable Resource hashing function
-func (a *API) ResourceHashSize() int {
-	return a.resource.HashSize
+// FeedsHashSize returned the size of the digest produced by Swarm Feeds' hashing function
```
Comment (zelig) [inline swarm/api/api.go]: feeds'

File: swarm/api/client/client.go [hunk 19]
```diff
@@ -701,18 +700,18 @@ func (c *Client) getResource(query *mru.Query, manifestAddressOrDomain string, m
...
-		return nil, fmt.Errorf("Error retrieving resource: %s", errorMessage)
+		return nil, fmt.Errorf("Error retrieving feed updates: %s", errorMessage)
 	}
 
 	return res.Body, nil
 }
 
-// GetResourceMetadata returns a structure that describes the Mutable Resource
-// manifestAddressOrDomain is the address you obtained in CreateResource or an ENS domain whose Resolver
+// GetFeedRequest returns a structure that describes the referenced Feed status
```
Comment (zelig) [inline swarm/api/client/client.go]: feed status

File: swarm/api/http/server_test.go [hunk 20]
```diff
@@ -206,8 +165,8 @@ func TestBzzResourceMultihash(t *testing.T) {
 	}
 }
 
-// Test resource updates using the raw update methods
-func TestBzzResource(t *testing.T) {
+// Test Swarm Feeds using the raw update methods
```
Comment (zelig) [inline swarm/api/http/server_test.go]: feeds

File: swarm/network/README.md [hunk 21]
```diff
@@ -57,7 +57,7 @@ receipts for a deleted chunk easily to refute their challenge.
 - syncing should be resilient to cut connections, metadata should be persisted that
 keep track of syncing state across sessions, historical syncing state should survive restart
 - extra data structures to support syncing should be kept at minimum
-- syncing is organized separately for chunk types (resource update v content chunk)
+- syncing is organized separately for chunk types (Swarm Feed Updates v regular content chunk)
```
Comment (zelig) [inline swarm/network/README.md]: this is not true. it is handled not separately

File: swarm/pss/notify/notify.go [hunk 22]
```diff
@@ -317,7 +317,7 @@ func (c *Controller) handleStartMsg(msg *Msg, keyid string) (err error) {
 		return err
 	}
 
-	// TODO this is set to zero-length byte pending decision on protocol for initial message, whether it should include message or not, and how to trigger the initial message so that current state of MRU is sent upon subscription
+	// TODO this is set to zero-length byte pending decision on protocol for initial message, whether it should include message or not, and how to trigger the initial message so that current state of Swarm Feed is sent upon subscription
```
Comment (zelig) [inline swarm/pss/notify/notify.go]: feed

File: swarm/storage/feeds/error.go [hunk 23]
```diff
@@ -35,7 +35,7 @@ const (
 	ErrCnt
 )
 
-// Error is a the typed error object used for Mutable Resources
+// Error is a the typed error object used for Swarm Feeds
```
Comment (zelig) [inline swarm/storage/feeds/error.go]: feeds

File: swarm/storage/feeds/error.go [hunk 24]
```diff
@@ -47,12 +47,12 @@ func (e *Error) Error() string {
...
 
 // Code returns the error code
-// Error codes are enumerated in the error.go file within the mru package
+// Error codes are enumerated in the error.go file within the feeds package
 func (e *Error) Code() int {
 	return e.code
 }
 
-// NewError creates a new Mutable Resource Error object with the specified code and custom error message
+// NewError creates a new Swarm Feeds Error object with the specified code and custom error message
```
Comment (zelig) [inline swarm/storage/feeds/error.go]: feeds

File: swarm/storage/feeds/feed.go [hunk 25]
```diff
@@ -25,21 +25,21 @@ import (
 	"github.com/ethereum/go-ethereum/swarm/storage"
 )
 
-// View represents a particular user's view of a resource
-type View struct {
+// Feed represents a particular user's stream of updates on a Topic
```
Comment (zelig) [inline swarm/storage/feeds/feed.go]: topic

File: swarm/storage/feeds/feed.go [hunk 26]
```diff
@@ -48,55 +48,55 @@ func (u *View) mapKey() uint64 {
 	return *(*uint64)(unsafe.Pointer(&hash[0]))
 }
 
-// binaryPut serializes this View instance into the provided slice
-func (u *View) binaryPut(serializedData []byte) error {
-	if len(serializedData) != viewLength {
-		return NewErrorf(ErrInvalidValue, "Incorrect slice size to serialize View. Expected %d, got %d", viewLength, len(serializedData))
+// binaryPut serializes this Feed instance into the provided slice
```
Comment (zelig) [inline swarm/storage/feeds/feed.go]: feed

File: swarm/storage/feeds/feed.go [hunk 27]
```diff
@@ -48,55 +48,55 @@ func (u *View) mapKey() uint64 {
...
+func (f *Feed) binaryPut(serializedData []byte) error {
+	if len(serializedData) != feedLength {
+		return NewErrorf(ErrInvalidValue, "Incorrect slice size to serialize Feed. Expected %d, got %d", feedLength, len(serializedData))
```
Comment (zelig) [inline swarm/storage/feeds/feed.go]: feed

File: swarm/storage/feeds/feed.go [hunk 28]
```diff
@@ -48,55 +48,55 @@ func (u *View) mapKey() uint64 {
...
+	return feedLength
 }
 
 // binaryGet restores the current instance from the information contained in the passed slice
-func (u *View) binaryGet(serializedData []byte) error {
-	if len(serializedData) != viewLength {
-		return NewErrorf(ErrInvalidValue, "Incorrect slice size to read View. Expected %d, got %d", viewLength, len(serializedData))
+func (f *Feed) binaryGet(serializedData []byte) error {
+	if len(serializedData) != feedLength {
+		return NewErrorf(ErrInvalidValue, "Incorrect slice size to read Feed. Expected %d, got %d", feedLength, len(serializedData))
```
Comment (zelig) [inline swarm/storage/feeds/feed.go]: feed

File: swarm/storage/feeds/handler.go [hunk 29]
```diff
@@ -14,9 +14,9 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
-// Handler is the API for Mutable Resources
-// It enables creating, updating, syncing and retrieving resources and their update data
-package mru
+// Handler is the API for Feeds
```
Comment (zelig) [inline swarm/storage/feeds/handler.go]: feeds

File: swarm/storage/feeds/handler.go [hunk 30]
```diff
@@ -52,35 +52,35 @@ var hashPool sync.Pool
...
 		}
 		hashPool.Put(hashfunc)
 	}
 
-	return rh
+	return fh
 }
 
-// SetStore sets the store backend for the Mutable Resource API
+// SetStore sets the store backend for the Swarm Feeds API
```
Comment (zelig) [inline swarm/storage/feeds/handler.go]: feeds

File: swarm/storage/feeds/handler.go [hunk 31]
```diff
@@ -89,81 +89,78 @@ func (h *Handler) Validate(chunkAddr storage.Address, data []byte) bool {
...
 	}
 
 	return true
 }
 
-// GetContent retrieves the data payload of the last synced update of the Mutable Resource
-func (h *Handler) GetContent(view *View) (storage.Address, []byte, error) {
-	if view == nil {
-		return nil, nil, NewError(ErrInvalidValue, "view is nil")
+// GetContent retrieves the data payload of the last synced update of the Feed
```
Comment (zelig) [inline swarm/storage/feeds/handler.go]: of the feed

File: swarm/storage/feeds/handler.go [hunk 32]
```diff
@@ -267,32 +264,32 @@ func (h *Handler) Update(ctx context.Context, r *Request) (updateAddr storage.Ad
...
+	return feedUpdate
 }
 
-// Sets the resource cache value for the given View
-func (h *Handler) set(view *View, rsrc *cacheEntry) {
-	mapKey := view.mapKey()
-	h.resourceLock.Lock()
-	defer h.resourceLock.Unlock()
-	h.resources[mapKey] = rsrc
+// Sets the feed update cache value for the given Feed
```
Comment (zelig) [inline swarm/storage/feeds/handler.go]: given feed

File: swarm/storage/feeds/handler_test.go [hunk 33]
```diff
@@ -83,18 +83,18 @@ func TestResourceHandler(t *testing.T) {
...
 
-	rh, datadir, teardownTest, err := setupTest(clock, signer)
+	feedsHandler, datadir, teardownTest, err := setupTest(clock, signer)
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer teardownTest()
 
-	// create a new resource
+	// create a new Feed
```
Comment (zelig) [inline swarm/storage/feeds/handler_test.go]: feed

File: swarm/storage/feeds/handler_test.go [hunk 34]
```diff
@@ -83,18 +83,18 @@ func TestResourceHandler(t *testing.T) {
...
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	topic, _ := NewTopic("Mess with mru code and see what ghost catches you", nil)
-	view := View{
+	topic, _ := NewTopic("Mess with Swarm Feeds code and see what ghost catches you", nil)
```
Comment (zelig) [inline swarm/storage/feeds/handler_test.go]: feeds

File: swarm/api/http/server.go [hunk 35]
```diff
@@ -458,66 +457,35 @@ func (s *Server) HandleDelete(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, newKey)
 }
 
-// Parses a resource update post url to corresponding action
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply swarm/api/http/server.go]: old orphan code not used by any code any more. It has been sitting in there unused for months now.

File: swarm/api/http/server.go [hunk 36]
```diff
@@ -527,30 +495,30 @@ func (s *Server) HandlePostResource(w http.ResponseWriter, r *http.Request) {
...
+		// feed identification used to retrieve feed updates later
+		m, err := s.api.NewFeedManifest(r.Context(), &updateRequest.Feed)
 		if err != nil {
-			RespondError(w, r, fmt.Sprintf("failed to create resource manifest: %v", err), http.StatusInternalServerError)
+			RespondError(w, r, fmt.Sprintf("failed to create feed manifest: %v", err), http.StatusInternalServerError)
 			return
 		}
 		// the key to the manifest will be passed back to the client
-		// the client can access the view  directly through its resourceView member
+		// the client can access the Feed  directly through its Feed member
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply swarm/api/http/server.go]: fixed. Thanks.

File: .github/CODEOWNERS [hunk 37]
```diff
@@ -27,6 +27,6 @@ swarm/services                  @zelig
 swarm/state                     @justelad
 swarm/storage/encryption        @gbalint @zelig @nagydani
 swarm/storage/mock              @janos
-swarm/storage/mru               @nolash
+swarm/storage/feeds             @nolash
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply .github/CODEOWNERS]: done. Thanks.

File: swarm/network/README.md [hunk 38]
```diff
@@ -37,7 +37,7 @@ Using the streamer logic, various stream types are easy to implement:
   * live session syncing
   * historical syncing
 * simple retrieve requests and deliveries
-* mutable resource updates streams
+* Swarm Feeds streams
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply swarm/network/README.md]: fixed. thanks

File: swarm/storage/feeds/binaryserializer.go [hunk 39]
```diff
@@ -14,7 +14,7 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
-package mru
+package feeds
```
[the remark this answers is not in the document]
Comment (jpeletier) [reply swarm/storage/feeds/binaryserializer.go]: "Swarm Feeds" sounded better to me from a "marketing" standpoint, i.e., *I am using Swarm Feeds in my project so my users can collaborate"*

Also, when instantiating the `Handler`, plural seemed better: `fh:= new(feeds.Handler)` (the handler of all the feeds). Singular seemed like the handler of 1 feed.

I'll rename it, the "marketing name" in documentation does not have to be exactly 1:1 with the code name.

File: swarm/storage/feeds/handler_test.go (hunk 3 above)
Comment (jpeletier) [reply swarm/storage/feeds/handler_test.go]: fixed. thanks

File: swarm/network/README.md (hunk 21 above)
Comment (jpeletier) [reply swarm/network/README.md]: changed to: *syncing is **not** organized separately for chunk types (Swarm feed updates v regular content chunk)*

However, take into account I was just changing terms here, not really changing the meaning. `swarm/network` owner should be aware of this.

**Comment by jpeletier** [comment]
@zelig 
* Renamed package to `feed`
* Renamed all conflicting vars to `f` or `fd`
* Fixed all capitalized `Feed` + regex searched and found more, also fixed.
* Rebased.

File: .github/CODEOWNERS [hunk 40]
```diff
@@ -27,6 +27,6 @@ swarm/services                  @zelig
 swarm/state                     @justelad
 swarm/storage/encryption        @gbalint @zelig @nagydani
 swarm/storage/mock              @janos
-swarm/storage/mru               @nolash
+swarm/storage/feeds             @nolash @jpeletier
```
Comment (zelig) [inline .github/CODEOWNERS]: feed
:)

File: swarm/OWNERS [hunk 41]
```diff
@@ -22,5 +22,5 @@ swarm
 ├── storage ─────────────── ethersphere
 │   ├── encryption ──────── @gbalint, @zelig, @nagydani
 │   ├── mock ────────────── @janos
-│   └── mru ─────────────── @nolash
+│   └── feeds ───────────── @nolash, @jpeletier
```
Comment (zelig) [inline swarm/OWNERS]: feed

File: swarm/swarm.go [hunk 42]
```diff
@@ -50,7 +50,7 @@ import (
 	"github.com/ethereum/go-ethereum/swarm/state"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 	"github.com/ethereum/go-ethereum/swarm/storage/mock"
-	"github.com/ethereum/go-ethereum/swarm/storage/mru"
+	"github.com/ethereum/go-ethereum/swarm/storage/feed"
```
Comment (zelig) [inline swarm/swarm.go]: formatting! 
https://travis-ci.org/ethereum/go-ethereum/jobs/436008109#L511

Comment (jpeletier) [reply swarm/swarm.go]: fixed. Sorry.

File: .github/CODEOWNERS (hunk 40 above)
Comment (jpeletier) [reply .github/CODEOWNERS]: :+1:

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**